compiler/src/dotty/tools/dotc/interactive/Interactive.scala

```diff
@@ -96,18 +126,140 @@ object Interactive {
     .getOrElse((0, Nil))
   }
 
+  /** Get possible completions from tree at `pos`
+   *
+   *  @return offset and list of symbols for possible completions
+   */
+  def completions(pos: SourcePosition)(implicit ctx: Context): (Int, List[Symbol]) = {
+    val path = pathTo(ctx.compilationUnit.tpdTree, pos.pos)
+    computeCompletions(pos, path)(contextOfPath(path))
+  }
+
+  private def computeCompletions(pos: SourcePosition, path: List[Tree])(implicit ctx: Context): (Int, List[Symbol]) = {
+    val completions = Scopes.newScope.openForMutations
+
+    val (completionPos, prefix, termOnly, typeOnly) = path match {
+      case (ref: RefTree) :: _ =>
+        if (ref.name == nme.ERROR)
+          (ref.pos.point, "", false, false)
+        else
+          (ref.pos.point,
+           ref.name.toString.take(pos.pos.point - ref.pos.point),
+           ref.name.isTermName,
+           ref.name.isTypeName)
+      case _ =>
+        println(i"COMPUTE from ${path.headOption}")
+        (0, "", false, false)
+    }
+
+    /** Include in completion sets only symbols that
+     *   - start with given name prefix
+     *   - do not contain '$' except in prefix where it is explicitly written by user
+     *   - have same term/type kind as name prefix given so far
+     */
+    def include(sym: Symbol) =
+      sym.name.startsWith(prefix) &&
+      !sym.name.toString.drop(prefix.length).contains('$') &&
```

Comment:
Can't we just check if the name kind is SimpleName after a `stripModuleClassSuffix` ?

Reply:
I want to exclude all compile-generated names, not just module names.

Reply:
But all these names will have a kind different from SimpleName, no?